src/Command/ReferencesIndexCommand.php

```diff
@@ -89,7 +95,11 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $src = Path::canonicalize($input->getArgument('src'));
 
         foreach ($this->getFiles($src, $input->getOption('namespace'), (array) $input->getOption('exclude'), (array) $input->getOption('tags-to-ignore'), (array) $input->getOption('exclude-path')) as $class => $file) {
-            $class = new ClassParser(new ReflectionClass($class));
+            $class = new ClassParser(
+                reflection: new ReflectionClass($class),
+                url: str_replace($src, $input->getOption('base-url'), $file->getPathname())
```

Comment:
I don't think injecting the `url` argument in parsers is a good design. Parsers are designed to be created with a single Reflection object from any place. Adding this `url` argument will impose to carry it from parser to parser.

ClassParser objects are created:
- in Twig extensions
- in parsers: TypeParser, ClassParser, MethodParser
- in Commands

As the url is dedicated for rendering the view, converting the parser object as a view object (handling this `url` argument) and injecting it to the view might be better, as you suggested.